### PR TITLE
[CPU] Adopt DetectionOutput-1 to DetectionOutput-8

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Adopt the current implementation of *DetectionOutput* to *opset-8* specification, i.e. without *numClasses* attribute.

**Ticket:** 65164

Signed-off-by: Roman Kazantsev <roman.kazantsev@intel.com>

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_detection_output_node.cpp [hunk 1]
```diff
@@ -33,9 +33,9 @@ bool SortScorePairDescend<std::pair<int, int>>(const std::pair<float, std::pair<
 
 bool MKLDNNDetectionOutputNode::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept {
     try {
-        const auto doOp = ngraph::as_type_ptr<const ngraph::op::v0::DetectionOutput>(op);
+        const auto doOp = ngraph::as_type_ptr<const ngraph::op::v8::DetectionOutput>(op);
```
[the remark this answers is not in the document]
Comment (rkazants) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_detection_output_node.cpp]: Fixed. Thanks

**Review by nshchego — APPROVED** [review]

File: inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp [hunk 2]
```diff
@@ -374,11 +376,13 @@ static void TransformationUpToCPUSpecificOpSet(std::shared_ptr<ngraph::Function>
     pass_config->disable<ngraph::pass::ConvertReduceMaxToPooling>();
     pass_config->disable<ngraph::pass::ConvertReduceSumToPooling>();
     pass_config->disable<ngraph::pass::SliceToStridedSlice>();
+    pass_config->disable<ngraph::pass::ConvertDetectionOutput8ToDetectionOutput1>();
 
     pass_config->enable<ngraph::pass::NormalizeL2Decomposition>();
     pass_config->enable<ngraph::pass::ConvertInterpolate1ToInterpolate4>();
     pass_config->enable<ngraph::pass::ConvertGather1ToGather7>();
     pass_config->enable<ngraph::pass::ConvertGather8ToGather7>();
+    pass_config->enable<ngraph::pass::ConvertDetectionOutput1ToDetectionOutput8>();
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:385]: ConvertDetectionOutput1ToDetectionOutput8 also added in common optimizations. Looks like it is not required here.

File: inference-engine/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/builders.hpp [hunk 3]
```diff
@@ -481,6 +481,9 @@ std::shared_ptr<ngraph::Node> makeLogical(const ngraph::ParameterVector& inputs,
 std::shared_ptr<ngraph::Node> makeDetectionOutput(const ngraph::OutputVector &inputs,
                                                   const ngraph::op::DetectionOutputAttrs& attrs);
 
+std::shared_ptr<ngraph::Node> makeDetectionOutputV8(const ngraph::OutputVector &inputs,
```
Comment (nshchego) [inline inference-engine/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/builders.hpp:484]: ngraph->ov here and below.

File: inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp (hunk 2 above)
Comment (rkazants) [reply inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:383]: Reverted

File: inference-engine/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/builders.hpp (hunk 3 above)
Comment (rkazants) [reply inference-engine/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/builders.hpp]: These changes are reverted

**Comment by rkazants** [comment]
Now cpuFuncTests pass

File: src/common/transformations/src/transformations/op_conversions/detection_output_upgrade.cpp [hunk 4]
```diff
@@ -48,13 +48,13 @@ pass::ConvertDetectionOutput1ToDetectionOutput8::ConvertDetectionOutput1ToDetect
 
         std::shared_ptr<opset8::DetectionOutput> detection_output_v8_node = nullptr;
         if (detection_output_v1_node->get_input_size() == 3) {
-            auto detection_output_v8_node =
+            detection_output_v8_node =
```
Comment (nshchego) [inline src/common/transformations/src/transformations/op_conversions/detection_output_upgrade.cpp:51]: It should be covered by transformation tests. Actually transformation was not working and issue was detected during enabling of transformation in the plug-in.

Comment (rkazants) [reply src/common/transformations/src/transformations/op_conversions/detection_output_upgrade.cpp:51]: I will investigate it

Comment (rkazants) [reply src/common/transformations/src/transformations/op_conversions/detection_output_upgrade.cpp:51]: @nshchego, investigated. There is a gap in type_info comparator used in transformations tests. I emailed about this bug in detail. For now, I use more advanced compator to check attributes as well which number is changed from v1 to v8 opset. So now the transformation tests are more reliable.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**